spec/models/endpoint_spec.rb

```diff
@@ -45,4 +45,16 @@
       end
     end
   end
+
+  context "Uniqueness validation on :url" do
+    it "is not required" do
+      expect(Endpoint.create!).to be_truthy
+      expect(Endpoint.create!).to be_truthy
+    end
+
+    it "raises when provided and already exists" do
+      Endpoint.create!(:url => "abc")
+      expect { Endpoint.create!(:url => "abc") }.to raise_error("Validation failed: Url has already been taken")
+    end
+  end
```

Comment:
TBH, I'm indifferent about these specs. You're essentially testing that ActiveRecord does it's job (which is also why I'm generally against [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers). Are you sure you want to add these db writes to the tests? Your choice but thought I'd throw my thoughts out there.